File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(string s)
{
    vector<char> v;
    int answer=0;
    while (s.size() > 0) {
        if (v.size() > 0 && v.back() == s.back()) {
            v.pop_back();
            s.pop_back();
        } else {
            v.push_back(s.back());
            s.pop_back();
        }
    }
    if(v.size() == 0) {
        answer = 1;
    }
    
    return answer;
}

File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = 0
    lst = list(s)
    stack = []
    while lst:
        c = lst.pop()
        if len(stack) > 0 and stack[-1] == c:
            stack.pop()
        else:
            stack.append(c)
    if not stack:
        answer = 1
        
    return answer